concent_api/concent_api/middleware.py

```diff
@@ -39,3 +50,80 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+def should_use_html(request):
+    """
+    Function inspects request headers and determines is response should be in HTML format.
+    1) By default it HTML should not be used
+    2) If `text/html` is present in 'Accept' header and 'application/json' is not, HTML should be used
+    3) If both are present, HTML will be used if it has greater weight than JSON
+    """
+    def get_probability(selected_method, accepted_methods):
+        for method in accepted_methods:
+            if selected_method in method:
+                if ":q=" in method:
+                    return float(method.split(":q=")[1])
```

Comment:
This will result in HTTP 500 when the weight is not something that can be converted to `float`. You should return some kind of 4xx error in that situation instead.

Basically, assume that the format is `<content_type>(;q=<number>)?`. Maybe allow white space between components but anything other than that should be treated as an error. For example weight that is not a number or repeating the `;q=` part twice.

Reply:
This is bascially checkin the format of request header. It should not be done in InternalServerErrorHandlingMiddleware. It should be validated on `process_request` step by different entity (middleware)

Reply:
Then there should be at least an `assert` for this. This middleware basically depends on that other middleware doing this. A dependency like this is something that will break very easily and no one will notice if we don't have a sanity check enforcing this.

If someone removes the other middleware in settings I think we would much prefer to get `AssertionError` than just for it to stop working silently. An `assert` is a quick and simple solution.